src/overlays/actors/ovl_Door_Spiral/z_door_spiral.c

```diff
@@ -31,30 +40,223 @@ const ActorInit Door_Spiral_InitVars = {
+    DoorSpiral* this = THIS;
+    s32 transitionActorId = DOORSPIRAL_GET_FC00(thisx);
+    s8 objectSlot;
+
+    if (this->actor.room != play->doorCtx.transitionActorList[transitionActorId].sides[0].room) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    this->type = DOORSPIRAL_GET_TYPE(thisx);
+    this->direction = DOORSPIRAL_GET_80(thisx);
+    this->unk147 = func_809A2BF8(play);
+    objectSlot = Object_GetSlot(&play->objectCtx, sSpiralObjectInfoTable[this->unk147].objectId);
+    this->objSlot = objectSlot;
+    if (objectSlot < 0) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+    DoorSpiral_SetupAction(this, func_809A2DB0);
+    Actor_SetFocus(&this->actor, 60.0f);
+}
+
+void DoorSpiral_Destroy(Actor* thisx, PlayState* play) {
+    s32 transitionActorId = DOORSPIRAL_GET_FC00(thisx);
+
+    play->doorCtx.transitionActorList[transitionActorId].id = -play->doorCtx.transitionActorList[transitionActorId].id;
+}
+
+void func_809A2DB0(DoorSpiral* this, PlayState* play) {
+    if (Object_IsLoaded(&play->objectCtx, this->objSlot)) {
+        this->actor.objectSlot = this->objSlot;
+        func_809A2B70(this, play);
+    }
+}
+
+f32 func_809A2E08(PlayState* play, DoorSpiral* this, f32 arg2, f32 arg3, f32 arg4) {
+    Player* player = GET_PLAYER(play);
+    Vec3f point;
+    Vec3f offset;
+
+    point.x = player->actor.world.pos.x;
+    point.y = player->actor.world.pos.y + arg2;
+    point.z = player->actor.world.pos.z;
+
+    Actor_OffsetOfPointInActorCoords(&this->actor, &offset, &point);
+
+    if ((arg3 < fabsf(offset.x)) || (arg4 < fabsf(offset.y))) {
+        return FLT_MAX;
+    } else {
+        return offset.z;
+    }
+}
+
+s32 func_809A2EA0(DoorSpiral* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    if (!Player_InCsMode(play)) {
+        SpiralInfo* temp_v0 = &sSpiralInfoTable[this->unk148];
+        f32 temp_fv0 = func_809A2E08(play, this, 0.0f, temp_v0->unk_0B, temp_v0->unk_0C);
+
+        if (fabsf(temp_fv0) < 64.0f) {
+            s16 var_v0 = player->actor.shape.rot.y - this->actor.shape.rot.y;
+
+            if (temp_fv0 > 0.0f) {
+                var_v0 = 0x8000 - var_v0;
+            }
+            if (ABS_ALT(var_v0) < 0x3000) {
+                return (temp_fv0 >= 0.0f) ? 1.0f : -1.0f;
+            }
+        }
+    }
+    return 0;
+}
+
+void func_809A2FF8(DoorSpiral* this, PlayState* play) {
+    Player* player;
+    u32 index;
+
+    if (this->shouldClimb) {
+        DoorSpiral_SetupAction(this, func_809A3098);
+    } else if (func_809A2EA0(this, play) != 0) {
+        player = GET_PLAYER(play);
+        player->doorType = PLAYER_DOORTYPE_STAIRCASE;
+        player->doorDirection = this->direction;
+        player->doorActor = &this->actor;
+        index = DOORSPIRAL_GET_FC00(&this->actor);
+        player->doorNext = (play->doorCtx.transitionActorList[index].params >> 0xA);
```

Comment:
```suggestion
        player->doorNext = (play->doorCtx.transitionActorList[transitionId].params >> 0xA);
```
Also, should this take `DOOR_GET_TRANSITION_ID`? (would like others thoughts on this too)